### PR TITLE
Add a test suite for ConfigFile

### PR DESCRIPTION
**Pull request by Calinou**
It can likely be completed further, but this should be a good start.

### CODE REVIEW

File: tests/test_config_file.h [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+			String(config_file.get_value("player", "tagline")) == "Waiting\nfor\nGodot",
+			"Reading `player/tagline` should return the expected value.");
+	CHECK_MESSAGE(
+			Color(config_file.get_value("player", "color")).is_equal_approx(Color(0, 0.5, 1)),
+			"Reading `player/color` should return the expected value.");
+	CHECK_MESSAGE(
+			Vector2(config_file.get_value("player", "position")).is_equal_approx(Vector2(3, 4)),
+			"Reading `player/position` should return the expected value.");
+	CHECK_MESSAGE(
+			bool(config_file.get_value("graphics", "antialiasing")) == true,
```
Comment (akien-mga) [inline tests/test_config_file.h:79]: `misc/scripts/file_format.sh` doesn't like redundant `== true` (though here I tend to disagree with the script).

Comment (Xrayez) [reply tests/test_config_file.h:79]: I think the conditional can be safely moved to a variable up above before the assertion.
```c++
const bool config_value_enabled = config_file.get_value("graphics", "antialiasing")
CHECK_MESSAGE(config_value_enabled, "Reading `graphics/antialiasing` should return the expected value.")
```
Or `antialiasing_enabled`, but may be misleading to the purpose of testing the `ConfigFile` specifically, or combine both...

Or add static check exceptions for `tests/`. 🙂

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Needs a rebase.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**